pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,172 @@
+// named "cluster" or a configmap object in namespace "openshift-config".
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	loggingNamespacedName := types.NamespacedName{Name: singletonName, Namespace: openshiftNS}
+	proxyNamespacedName := types.NamespacedName{Name: proxyName}
+	if request.NamespacedName == proxyNamespacedName {
+		proxyConfig := &configv1.Proxy{}
+		logrus.Debugf("Proxyconfig reconcile request.Name: '%s'", request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("proxy not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, err
+		}
+	} else if request.Name == singletonName {
+		trustBundle := &corev1.ConfigMap{}
+		logrus.Debugf("Trust bundle configmap reconcile request.Namespace/request.Name: '%s/%s'", request.Namespace, request.Name)
+		if err := r.client.Get(context.TODO(), loggingNamespacedName, trustBundle); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("configmap not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, err
+		}
+	} else {
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch the ClusterLogging instance
+	instance := &loggingv1.ClusterLogging{}
+	if err := r.client.Get(context.TODO(), loggingNamespacedName, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - just return without requeuing.
+		return reconcile.Result{}, nil
+	}
+
+	if instance.Spec.ManagementState == loggingv1.ManagementStateUnmanaged {
+		return reconcile.Result{}, nil
+	}
+
+	forwardinginstance := &logforwarding.LogForwarding{}
+	err := r.client.Get(context.TODO(), loggingNamespacedName, forwardinginstance)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// Request object not found, could have been deleted after reconcile request.
+		// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	if err := k8shandler.ReconcileForGlobalProxy(instance, forwardinginstance, r.client); err != nil {
+		// Return and don't requeue
+		return reconcile.Result{}, nil
```

Comment:
i think we do want to requeue here... if we fail during us trying to process something we should try to ensure it sticks